server/server_test.go

```diff
@@ -45,11 +81,14 @@ func TestRunAndShutdown(t *testing.T) {
 		t.Fatalf("Failed to create server: %v", err)
 	}
 
+	shutdownCh := make(chan struct{})
 	go func() {
-		time.Sleep(time.Second)
+		<-shutdownCh
 		server.Shutdown()
```

Comment:
It's better than `time.Sleep`, but is not entirely correct. Try adding a panic("hi") in the last line of `Shutdown()` to see the test doesn't catch the panic